doc: update release notes for v1.9

Shilpashree3lc-zz opened the pull request:
Added the list of Jira items, Kernel and HAL related changes for release notes v1.9

Signed-off-by: Shilpashree L C <shilpashree.lc@intel.com>

dbkinder reviewed (changes requested):
Please check https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/3845/3/tests

Commit message has lines that are too long

Also, the list of Jira issues will need updating when we actually get to the release point.  Curious what the criteria is for this list:  I notice that ZEP-2229 is not listed, but it's marked as closed/fixed for 1.9 (and it's not listed in the 1.8 release notes).

Shilpashreelc3 committed:
doc: update release notes for v1.9

Added the list of Jira items, Kernel and HAL
related changes for release notes v1.9

Signed-off-by: Shilpashree L C <shilpashree.lc@intel.com>

Shilpashree3lc commented:
I have done the changes. ZEP-2229 is not listed because it was closed on 02/Jun/17 but the fix version mention was v1.9. Since the filter I was using couldn't recognize it because we were using date closed after 2017-06-15. I have now added this to the list.
